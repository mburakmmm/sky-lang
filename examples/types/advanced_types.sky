# Advanced types test

function add(a: int, b: int): int
  return a + b
end

function main
  print("=== List Type Test ===")
  let numbers: [int] = [1, 2, 3, 4, 5]
  print(len(numbers))
  print(numbers[0])
  print(numbers[4])
  
  print("=== Dict Type Test ===")
  let person = {"name": "Alice", "age": "30", "city": "NYC"}
  print(person["name"])
  print(person["age"])
  
  print("=== Nested Types Test ===")
  let matrix: [[int]] = [[1, 2], [3, 4], [5, 6]]
  print(matrix[0][0])
  print(matrix[1][1])
  print(matrix[2][0])
  
  print("=== Function Type Test ===")
  let result = add(10, 20)
  print(result)
  
  print("=== Done ===")
end

