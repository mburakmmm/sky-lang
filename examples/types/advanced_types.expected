=== List Type Test ===
5
1
5
=== Dict Type Test ===
Alice
30
=== Nested Types Test ===
1
4
5
=== Function Type Test ===
30
=== Done ===

